### ISSUE TITLE
Struct Deconstruct with cfg is reported as erroneous 

### ISSUE THREAD

**Issue by niko-renko** [rust-lang/rust-analyzer#16169]
```rust
struct Test {
    #[cfg(feature = "hello")]
    test: u32,
    other: u32
}

fn main() {
    let a = Test {
        #[cfg(feature = "hello")]
        test: 1,
        other: 1
    };

    let Test {
        #[cfg(feature = "hello")]
        test,
        mut other,
        ..
    } = a;

    other += 1;
}
```
rust-analyzer reports [E0559](https://doc.rust-lang.org/stable/error_codes/E0559.html) in deconstruction in the snippet above. 

**rust-analyzer version**: 0.3.1774-standalone

**rustc version**: 1.74.1 (a28077b28 2023-12-04)

**relevant settings**: N/A
